projects/plugins/jetpack/modules/masterbar/masterbar/class-masterbar.php

```diff
@@ -104,13 +104,16 @@ public function __construct() {
 		$this->user_email      = $this->user_data['email'];
 		$this->display_name    = $this->user_data['display_name'];
 		$this->user_site_count = $this->user_data['site_count'];
+		$this->is_rtl          = 'rtl' === $this->user_data['text_direction'];
 
 		// Store part of the connected user data as user options so it can be used
 		// by other files of the masterbar module without making another XMLRPC
 		// request. Although `get_connected_user_data` tries to save the data for
 		// future uses on a transient, the data is not guaranteed to be cached.
 		if ( isset( $this->user_data['use_wp_admin_links'] ) ) {
-			update_user_option( get_current_user_id(), 'jetpack_admin_menu_link_destination', $this->user_data['use_wp_admin_links'] );
+			$user_id = get_current_user_id();
+			update_user_option( $user_id, 'jetpack_admin_menu_link_destination', $this->user_data['use_wp_admin_links'] );
+			update_user_option( $user_id, 'jetpack_wpcom_is_rtl', $this->is_rtl );
```

Comment:
Should we move this to a separate check instead of `isset( $this->user_data['use_wp_admin_links'] )` ?

Reply:
Oops, good catch! Fixed in https://github.com/Automattic/jetpack/pull/19317/commits/f780d4e1997a876a97b3a98172ec0c0eb29dac7b.